CountingJS/script.js

```diff
@@ -0,0 +1,35 @@
+let i = 1;
+
+for (let i = 1; i <= 10; i++) {
+	console.log(i);
+  }
+for (let i = 10; i >= 1; i--) {
+	console.log(i);
+}
+
+while (i <= 10) {
+	console.log(i);
+	i++
+  }
+while (i >= 1) {
+	console.log(i);
+	i--
+}
+
+do{
+	console.log(i);
+	i++
+  }
+while (i < 10);
+do{
+	console.log(i);
+	i--
+}
+while (i >= 1) ;
+
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+let text = "";
+for(let x of numbers){
+	text += x;
+	console.log(text);
+}
```

Comment:
This could also be:

```suggestion
const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
for (const number of numbers) {
    console.log(number);
}
```

1. Notice the spaces used as syntax on the first line of the loop
2. Prefer `const` over `let`
3. The variable naming convention in a `for...of` loop is the singular form of the array name being looped over
4. You can reference this variable in the loop body, it's in the scope. So just `console.log(number)` directly instead of making and updating a string, and printing that string.